File: Discord_win/Discord_win/Tools/JSONBuilder.cs
using Discord_win.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Discord_win.Tools {
    static class JsonBuilder {
        public static string BuildJsonFromHashMap(Dictionary<string, string> parameters) {
            StringBuilder resultJson = new StringBuilder();
            resultJson.Append("{");
            foreach (var item in parameters) {
                resultJson.Append("\"");
                resultJson.Append(item.Key);
                resultJson.Append("\"");
                resultJson.Append(":");
                resultJson.Append("\"");
                resultJson.Append(item.Value);
                resultJson.Append("\"");
                resultJson.Append(",");
            }
            resultJson.Remove(resultJson.Length - 1, 1);
            resultJson.Append("}");
            return resultJson.ToString();
        }
        public static string BuildLoginJson(string email, string password) {
            Dictionary<string, string> parameters = new Dictionary<string, string>();
            parameters.Add("email", email);
            parameters.Add("password", password);
            return BuildJsonFromHashMap(parameters);
        }
        public static string BuildMessageJson(int channelId, int userId, string content) {
            Dictionary<string, string> parameters = new Dictionary<string, string>();
            parameters.Add("channelId", channelId.ToString());
            parameters.Add("userId", userId.ToString());
            parameters.Add("content", content);
            parameters.Add("time", DateTime.Now.ToString("yyyy-MM-dd'T'HH:mm:ss.fff"));
            return BuildJsonFromHashMap(parameters);
        }
        public static string BuildServerJson(int userId, string name) {
            Dictionary<string, string> parameters = new Dictionary<string, string>();
            parameters.Add("adminId", userId.ToString());
            parameters.Add("name", name);
            return BuildJsonFromHashMap(parameters);
        }
        public static string BuildUserJson(string email, string password, string userName, string firstName, string lastName, Gender gender) {
            Dictionary<string, string> parameters = new Dictionary<string, string>();
            parameters.Add("email", email);
            parameters.Add("password", password);
            parameters.Add("userName", userName);
            parameters.Add("firstName", firstName);
            parameters.Add("lastName", lastName);
            parameters.Add("gender", gender.ToString());
            return BuildJsonFromHashMap(parameters);
        }
    }
}

File: Discord_android/Discord/app/src/main/java/com/peanut/discord/tools/JsonBuilder.java
package com.peanut.discord.tools;

import com.peanut.discord.MainActivity;
import com.peanut.discord.models.Channel;
import com.peanut.discord.models.User;


import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.HashMap;
import java.util.Iterator;
import java.util.Map;

public class JsonBuilder {
    private static final String dateTimePattern = "yyyy-MM-dd'T'HH:mm:ss.SSSSSSS";
    private String buildJsonFromHashMap(HashMap<String, String> parameters){
        StringBuilder resultJSON = new StringBuilder();
        Iterator iterator = parameters.entrySet().iterator();
        Map.Entry pair;
        resultJSON.append("{");
        while (iterator.hasNext()) {
            pair = (Map.Entry) iterator.next();
            resultJSON.append("\"");
            resultJSON.append(pair.getKey());
            resultJSON.append("\"");
            resultJSON.append(":");
            resultJSON.append("\"");
            resultJSON.append(pair.getValue());
            resultJSON.append("\"");
            resultJSON.append(",");
        }
        resultJSON.deleteCharAt(resultJSON.length() - 1);
        resultJSON.append("}");
        return resultJSON.toString();
    }
    public String buildLoginJson(String email, String password){
        HashMap<String, String> parameters = new HashMap<>();
        parameters.put("email", email);
        parameters.put("password", password);
        return buildJsonFromHashMap(parameters);
    }
    public String buildMessageJson(Channel currentChannel, User currentUser, String content){
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(dateTimePattern, MainActivity.locale);
        Date time = new Date(System.currentTimeMillis());
        HashMap<String, String> parameters = new HashMap<>();
        parameters.put("channelId", String.valueOf(currentChannel.getChannelId()));
        parameters.put("userId", String.valueOf(currentUser.getUserId()));
        parameters.put("content", content);
        parameters.put("time", simpleDateFormat.format(time));
        return buildJsonFromHashMap(parameters);
    }
    public String buildServerJson(int userId, String name){
        HashMap<String, String> parameters = new HashMap<>();
        parameters.put("adminId", String.valueOf(userId));
        parameters.put("name", name);
        return buildJsonFromHashMap(parameters);
    }
}
